industrial_robot_client/src/joint_trajectory_action.cpp

```diff
@@ -123,6 +191,24 @@ void JointTrajectoryAction::goalCB(JointTractoryActionServer::GoalHandle gh)
     return;
   }
 
+  // check robot can actually execute trajectory, if not, refuse goal.
+  // no point in accepting the goal only to cancel it immediately later
+  if (!is_motion_server_ok(last_robot_status_, consider_status_unknowns_ok_) && !ignore_motion_server_error_)
+  {
+    // translate status into user readable description
+    const std::string reject_msg = {"Rejecting goal: "
+      + describe_robot_status_msg(last_robot_status_, consider_status_unknowns_ok_) };
+    ROS_ERROR_STREAM_NAMED(name_, reject_msg);
+    control_msgs::FollowJointTrajectoryResult rslt;
+    // the goal is actually probably OK, but we have to choose one of the existing
+    // constants, and this one comes closest
+    rslt.error_code = control_msgs::FollowJointTrajectoryResult::INVALID_GOAL;
```

Comment:
we could also set the `rslt.error_string` ... not sure if anybody displays that though ... but I could imagine implementing that at some point in MoveIt

Reply:
`setRejected(..)` does that for us: [api doc](https://docs.ros.org/en/melodic/api/actionlib/html/classactionlib_1_1ServerGoalHandle.html#af0a4d2debf5b19d985553903ddaef92c).